File: bidappdemoapp-java/app/src/main/java/com/bidapp/demo/FullscreenLoadDelegate.java
package com.bidapp.demo;

import androidx.annotation.Nullable;

import io.bidapp.sdk.AdInfo;
import io.bidapp.sdk.BIDFullscreenLoadDelegate;

public class FullscreenLoadDelegate implements BIDFullscreenLoadDelegate {



    @Override
    public void didFailToLoad(AdInfo adInfo, Error error) {
        String waterfallId = adInfo != null ? adInfo.getWaterfallId() : "";
        String descr = error.getLocalizedMessage();
        System.out.println("Bidapp fullscreen " + waterfallId + " didFailToLoadAd: " + (adInfo != null ? adInfo.getNetworkId() : "null") + " " + adInfo + ". ERROR: " + descr);
    }

    @Override
    public void didLoad(@Nullable AdInfo adInfo) {
        String waterfallId = adInfo != null ? adInfo.getWaterfallId() : "";
        String isRewardedString = adInfo != null && adInfo.getAdFormat() != null ? Boolean.toString(adInfo.getAdFormat().isRewarded()) : "null";
        System.out.println("Bidapp fullscreen " + waterfallId + " didLoadAd: " + (adInfo != null ? adInfo.getNetworkId() : "null") + " " + adInfo + ". IsRewarded: " + isRewardedString);
    }
}

File: bidappdemoapp-compose/app/src/main/java/com/bidapp/demo/FullscreenLoadDelegate.kt
package com.bidapp.demo

import io.bidapp.sdk.AdInfo
import io.bidapp.sdk.BIDFullscreenLoadDelegate

class FullscreenLoadDelegate() : BIDFullscreenLoadDelegate {
    override fun didLoad(adInfo: AdInfo?) {
        val waterfallId = adInfo?.getWaterfallId() ?: ""
        val isRewardedString = adInfo?.adFormat?.isRewarded().toString() ?: "null";
        print("Bidapp fullscreen $waterfallId didLoadAd: ${adInfo?.getNetworkId()} $adInfo. IsRewarded: $isRewardedString")
    }

    override fun didFailToLoad(adInfo: AdInfo?, error: Error) {
        val waterfallId = adInfo?.getWaterfallId() ?: ""
        val descr = error.localizedMessage
        print("Bidapp fullscreen $waterfallId didFailToLoadAd: ${adInfo?.getNetworkId()} $adInfo. ERROR: $descr");
    }

}